core/templates/dev/head/pages/exploration_editor/statistics_tab/LearnerActionRenderService.ts

```diff
@@ -66,13 +67,17 @@ oppia.factory('LearnerActionRenderService', [
      * @param {int} timeSpentInStateSecs.
      * @param {string} currentStateName.
      * @param {int} actionIndex.
+     * @param {string} interactionId.
      * @returns {string}
      */
     var renderAnswerSubmitActionHTML = function(
         answer, destStateName, timeSpentInStateSecs, currentStateName,
-        actionIndex) {
+        actionIndex, interactionId) {
       var htmlString;
       if (currentStateName === destStateName) {
+        if (interactionId === 'FractionInput') {
```

Comment:
Hi @pranavsid98 -- I'm afraid this isn't really extensible. Can you find a way to do this that does not require special-casing a particular object factory instead?

For example, you might look at the interaction ID for the state, and use that to construct the correct "renderer" service that you can call to get the correct representation. But this should be something that's designed in such a way that it generalizes cleanly to other interactions.

Reply:
+1, I think it'd be nicer to have a service that maps interaction ids to the Factory that can build them, perhaps even programmatically (`'FractionInput' => $injector.get('Fraction' + 'ObjectFactory')`)?